File: app/src/main/java/com/kakacat/minitool/expressinquiry/Presenter.java
package com.kakacat.minitool.expressinquiry;

import android.content.Context;
import android.text.TextUtils;

import com.kakacat.minitool.common.myinterface.HttpCallback;
import com.kakacat.minitool.common.util.SystemUtil;
import com.kakacat.minitool.common.util.ThreadUtil;
import com.kakacat.minitool.expressinquiry.model.Delivery;
import com.kakacat.minitool.expressinquiry.model.Model;

import java.util.List;

import okhttp3.Response;

public class Presenter implements Contract.Presenter {

    private Contract.View view;
    private Model model;
    private Context context;

    public Presenter(Contract.View view) {
        this.view = view;
        this.model = Model.getInstance();
        this.context = view.getContext();
    }

    @Override
    public void initData() {
        model.initData(context);
    }

    @Override
    public void saveData() {
        model.saveDataToLocal(context);
    }

    @Override
    public void requestData(String code) {
        if (TextUtils.isEmpty(code)) {
            view.onRequestCallback("输入为空", false);
        } else if (!model.checkInput(code)) {
            view.onRequestCallback("输入错误,请检查输入", false);
        } else {
            if (model.isInSignedList(code)) {
                view.onRequestCallback("已经在签收列表哟", false);
            } else {
                model.sendRequest(code, new HttpCallback() {
                    @Override
                    public void onSuccess(Response response) {
                        if (model.handleResponse(response, code)) {
                            ThreadUtil.callInUiThread(() -> view.onRequestCallback("查询成功", true));
                        } else {
                            ThreadUtil.callInUiThread(() -> view.onRequestCallback("处理错误", false));
                        }
                    }

                    @Override
                    public void onError() {
                        ThreadUtil.callInUiThread(() -> view.onRequestCallback("请求错误", false));
                    }
                });
            }
        }
    }

    @Override
    public void refreshAll() {
        List<Delivery> list = getDeliveryList(0);
        if (list.size() == 0) {
            ThreadUtil.callInUiThread(() -> view.onRequestCallback("全部签收了哟", false));
        } else {
            list.forEach(delivery -> requestData(delivery.getCode()));
        }
    }

    @Override
    public CharSequence getDataFromClipBoard() {
        return SystemUtil.getDataFormClipBoard(context);
    }

    @Override
    public List<Delivery> getDeliveryList(int index) {
        return model.getList(index);
    }
}

File: app/src/main/java/com/kakacat/minitool/expressinquiry/Presenter.kt
package com.kakacat.minitool.expressinquiry

import com.kakacat.minitool.common.util.SystemUtil.getDataFormClipBoard
import com.kakacat.minitool.expressinquiry.model.Delivery
import com.kakacat.minitool.expressinquiry.model.Model
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.util.function.Consumer

class Presenter(private val view: Contract.View) : Contract.Presenter {

    private val model by lazy { Model() }
    private val context by lazy { view.context }

    override fun initData() {
        model.initData(context)
    }

    override fun saveData() {
        model.saveDataToLocal(context)
    }

    override fun requestData(code: String?) {
        GlobalScope.launch {
            if (!model.validateInput(code!!)) {
                view.onRequestCallback("输入错误,请检查输入")
            } else {
                if (model.isInSignedList(code)) {
                    view.onRequestCallback("已经在签收列表哟")
                } else {
                    val response = model.sendRequest(code)
                    val handleResult = model.handleResponse(response!!,code)
                    if(handleResult){
                        view.onRequestCallback("查询成功", true)
                    }else{
                        view.onRequestCallback("错误")
                    }
                }
            }
        }
    }

    override fun refreshAll() {
        val list = getDeliveryList(0)
        if (list!!.isEmpty()) {
            view.onRequestCallback("全部签收了哟")
        } else {
            list.forEach(Consumer { delivery: Delivery? -> requestData(delivery!!.code) })
        }
    }

    override val dataFromClipBoard: CharSequence
        get() = getDataFormClipBoard(context)

    override fun getDeliveryList(index: Int): List<Delivery>? {
        return model.getList(index)
    }
}